### PR TITLE
feat:done home-work

### PR DESCRIPTION
**Pull request by alma-amirseitov**

### CODE REVIEW

File: cat.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
+<?php
+require 'vendor/autoload.php';
+use GuzzleHttp\Client;
+
+function get_img_url($post)
+{
+    $client = new Client([
+        'base_uri' => 'https://api.thecatapi.com/v1/',
```
Comment (apushevye) [inline cat.php:8]: 'base_uri', 'timeout',  $uri 
Значения вышеперечисленных предлагаю записывать в константы. А здесь обращаться к константам. Таким образом не нужно будет искать по коду эти значения и проще будет их переиспользовать при необходимости.

File: cat.php [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        'timeout' => 5.0,
+    ]);
+    $uri = 'images/search';
+    $category = '';
```
Comment (apushevye) [inline cat.php:12]: необязательная строчка

File: cat.php [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    if (isset($post['category'])){
```
Comment (apushevye) [inline cat.php:13]: можно сократить до $category = $post['category'] ?? '';

File: cat.php [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+            break;
+    }
+    try {
+        $response = $client->request('GET', $uri);
+        $content = $response->getBody()->getContents();
+
+        return json_decode($content,true)[0]['url'];
+    }catch (\GuzzleHttp\Exception\GuzzleException $e) {
+        $response = $e->getResponse();
+        return "Server Error";
```
Comment (apushevye) [inline cat.php:48]: тут лучше выкидывать ошибку с контекстом, чтобы было понимание в чем дело.
Еще лучше добавить какую-нибудь библиотеку для логирования и писать более подробную информацию.

File: cat.php [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+            $uri .= '?category_ids=7';
+            break;
+        default:
+            break;
+    }
+    try {
+        $response = $client->request('GET', $uri);
+        $content = $response->getBody()->getContents();
+
+        return json_decode($content,true)[0]['url'];
```
Comment (apushevye) [inline cat.php:45]: Подумай что случится если апи будет возвращать 500. Не сломается ли у тебя ничего?

File: index.php [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
+    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.2.1/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-iYQeCzEYFbKjA/T2uDLTpkwGzCiq6soy8tYaI1GyVh/UjpbCx/TYkiZhlZB6+fzT" crossorigin="anonymous">
+    <title>Cat</title>
+</head>
+<body>
+<div class="container text-center mt-5">
+    <form id="form" action="index.php" method="post">
+        <fieldset>
+            <div class="mb-3">
+                <select name="category" class="form-select" aria-label="Default select example">
```
Comment (apushevye) [inline index.php:15]: Было бы круче добавить получение категории из апи, благо такая возможность предоставлена thecatapi =)

File: index.php [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+                    <option name="category" selected>Select category</option>
```
Comment (apushevye) [inline index.php:16]: Кстати, тут же можно добавить объекты типа Category заполнять ему параметры id и label например

File: theory.txt [hunk 8]
```diff
@@ -0,0 +1,22 @@
+Ping -команда позоляющая проверку соединените между компютерами или сервером
+
+при запуске команды ping google.com:
+ - число отправленных байтов,
+ - выводяться инвормация о IP адреса с кем мы хоти проверить соединение,
+ - icmp_seq номер последовательности
```
Comment (apushevye) [inline theory.txt:6]: Можно немного дополнить про icmp

File: theory.txt [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+ - время потраченное на отклик,
+после завершение команды, выводится информация о пакетах отпралена/доставлена/потерия
+
+---------------------------------
+Утилита curl используется для взаимодействия с серверами доставляя или получая даннные через разные протолы и методы
+
+запуская команду `curl -I https://kolesa.kz/` мы отправляем GET запрос на указанный url,
+опция -I означает что мы хотим получить только response Header(загаловки ответа)
+Полученные данные:
+- версия HTTP и статус ответа (HTTP/2 200 - успешно)
```
Comment (apushevye) [inline theory.txt:17]: Тут можно дополнить какие еще коды могут приходить и в чем их отличие

**Review by apushevye — CHANGES_REQUESTED** [review]
Работа принята, молодец! Но желательно доработать и сделать правки.

Раз уж мы прошли ООП, будет правильней его придерживаться. Как минимум можно создать объекты категорий и их заполнять.
Лучше разбить код. Писать все в index.php не есть хорошо. Если добавить ооп само собой исправится)
Можно немного дополнить теоретические ответы.
Пройтись по комментам оставленным в ПР
